### PR TITLE
Implement new readme generator

### PR DESCRIPTION
**Pull request by osterman**
## what
* Implement a new `README.md` generator.
* Use a [`README.yaml`](https://github.com/cloudposse/build-harness/blob/better-readme/README.yaml) datasource
* Support automatic inclusion of files in `docs/` (e.g. `targets.md` like in this repo)
* Drop support for recursion
* Add a `readme/lint` target to ensure `README.md` is up to date

## why
* Create a very easy format for defining consistent READMEs
* The `README.yaml` can be read by our `hugo` docs site to include specific sections in our documentation

## references
- <https://github.com/cloudposse/build-harness/blob/better-readme/README.md> 
- <https://github.com/cloudposse/build-harness/blob/better-readme/README.yaml> 

## design
* The design was a little bit influenced by hugo and it's use of "frontmatter"
* the `REAMDE.md` in this repo was generated from `templates/README.md` by using the data in `README.yaml` -- no other magic
* by default, `build-harness` will use the `templates/README.md` gomplate
* we can have different versions of this template. for now, this isi the only one.
* Recursion was removed b/c it leads to complicated templates that are difficult for others to understand
* Support for terraform `variable` and `output` documentation can be supported by writing their values to `docs/outputs.md` and `docs/variables.md` and then including them in the `includes` parameter in the `README.yaml` config

## todo

- [x] add a target to generate terraform variable/output documentation 
- [x] Link to documention hub
- [x] Related repos section
- [x] Logo support
- [x] link to open source project 
- [x] Related repos section
- [x] Link to jobs
- [x] Link to “paid support” page

### CODE REVIEW

File: templates/README.md [hunk 1]
```diff
@@ -1,16 +1,136 @@
...
+
+{{ range $file := (datasource "config").include -}}
+{{ (include "includes" $file) }}
+{{- end }}
+
+## Help
+
+**Got a question?**
+
+File a GitHub [issue]({{ printf "https://github.com/%s/issues" (ds "config").github_repo}}), send us an [email][email] or join our [Slack Community][slack]).
```
Comment (aknysh) [inline templates/README.md]: or join our [Slack Community][slack] - delete `)`

File: templates/README.md [hunk 2]
```diff
@@ -1,16 +1,136 @@
...
+- **Troubleshooting.** We'll help you triage why things aren't working.
+- **Code Reviews.** We'll review your Pull Requests and provide constructive feedback.
+- **Bug Fixes.** We'll rapidly work to fix any bugs in our projects.
+- **Build New Terraform Modules.** We'll develop original modules to provision infrastructure.
+- **Cloud Architecture.** We'll assist with your cloud strategy and design.
+- **Implementation.** We'll provide hands on support to implement our reference architectures. 
+
+## Community Forum
+
+Get access to our [Open Source Community Forum][slack] on Slack. It's **FREE** to join for everyone! Our "SweetOps" community is where you get to talk with others who share a similar vision for how to rollout and manage infrastructure. This is the best place to talk shop, ask questions, solict feedback, and work together as a community to build *sweet* infrastructure.
```
Comment (aknysh) [inline templates/README.md]: solicit feedback

File: templates/README.md [hunk 3]
```diff
@@ -1,16 +1,136 @@
...
+ 3. **Commit** changes to your own branch
+ 4. **Push** your work back up to your fork
+ 5. Submit a **Pull Request** so that we can review your changes
+
+**NOTE:** Be sure to merge the latest changes from "upstream" before making a pull request!
+
+{{ if eq (ds "config").license "APACHE2" }}
+## License
+
+[APACHE 2.0](LICENSE) © 2017 [Cloud Posse, LLC](https://cloudposse.com)
```
Comment (aknysh) [inline templates/README.md]: can we auto-generate the year here?
maybe a fixed start year and auto-generated current year if it's greater then the start year

**Review by aknysh — COMMENTED** [review]
+1
inline comments

File: templates/README.md (hunk 1 above)
Comment (osterman) [reply templates/README.md]: Too redundant?

File: templates/README.md (hunk 3 above)
Comment (osterman) [reply templates/README.md]: yea, good idea.

File: templates/README.md (hunk 1 above)
Comment (aknysh) [reply templates/README.md]: Just an unnecessary closing `)`

image

Comment (osterman) [reply templates/README.md]: thanks. good catch - i see.

File: README.yaml [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+# Name of this project
+name: Build Harness 
+
+# Logo for this project
+#logo: docs/logo.png
+
+# License of this project
+license: "APACHE2"
+
+# Cannonical GitHub repo
```
Comment (sarkis) [inline README.yaml]: Canonical

**Review by sarkis — APPROVED** [review]

**Pull request merged**